lab_3/main.py

```diff
@@ -222,7 +337,21 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        freq = {}
+        profile_as_dict = {}
+        for n_gram_trie in self.tries:
+            freq.update(((''.join(map(self.storage.get_letter_by_id, k)), v)
+                         for k, v in n_gram_trie.n_gram_frequencies.items()))
+        profile_as_dict["freq"] = freq
+        profile_as_dict["n_words"] = self.n_words
+        profile_as_dict["name"] = self.language
+        with open(name, "w", encoding="utf-8") as file:
```

Comment:
`file` is the Python built-in. Consider using another name.

Reply:
done